### PR TITLE
Added scroll functionality when naviagting lookup menu using keyboard

### PR DESCRIPTION
**Pull request by bereket-WMDE**
naviagting lookup menu using keyboard

Bug: T267829

### CODE REVIEW

File: vue-components/src/components/OptionsMenu.vue [hunk 1]
```diff
@@ -77,6 +77,13 @@ export default Vue.extend( {
 			type: Number,
 			default: -1,
 		},
+		/**
+		 * scroll lookup menu items into view while naviagitng using keyboard
+		 */
+		scrollWithKeyboard: {
```
Comment (micgro42) [inline vue-components/src/components/OptionsMenu.vue:83]: I'm not sure there will ever be a situation where we would not want to scroll with the keyboard. Which circumstances do you have in mind for this?

File: vue-components/src/components/OptionsMenu.vue [hunk 2]
```diff
@@ -106,6 +117,17 @@ export default Vue.extend( {
 					break;
 			}
 		},
+		keyboardScroll(): void {
+			const element = this.$refs[ 'menu-items' ] as HTMLElement[];
+
+			if ( this.scrollWithKeyboard && this.keyboardHoveredItemIndex !== -1 ) {
+				element[ this.keyboardHoveredItemIndex ].scrollIntoView( {
```
Comment (micgro42) [inline vue-components/src/components/OptionsMenu.vue:124]: And once you have removed the prop, I think you can fix the tests with this approach: https://stackoverflow.com/a/52446309/3293343

File: vue-components/src/components/OptionsMenu.vue (hunk 1 above)
Comment (bereket-WMDE) [reply vue-components/src/components/OptionsMenu.vue:83]: ah that was something i forgot to remove. i was thinking about the dropdown menu. but we do need it in that case as well

File: vue-components/src/components/OptionsMenu.vue (hunk 2 above)
Comment (bereket-WMDE) [reply vue-components/src/components/OptionsMenu.vue:124]: wow! thanks a lot for the tip

**Review by itamargiv — APPROVED** [review]
Great work, thank you!

**Pull request merged**